ocaml-lsp-server/src/document.ml

```diff
@@ -142,37 +145,43 @@ let make_config uri =
   Mconfig.get_external_config path mconfig
 
 let make_pipeline thread tdoc =
-  let async_make_pipeline =
-    Scheduler.async_exn thread (fun () ->
-        let text = Text_document.text tdoc in
-        let source = Msource.make text in
-        let config =
-          let uri = Text_document.documentUri tdoc in
-          make_config uri
+  ( lazy (Msource.make (Text_document.text tdoc))
+  , Lazy_fiber.create (fun () ->
+        let async_make_pipeline =
+          Scheduler.async_exn thread (fun () ->
+              let config =
+                let uri = Text_document.documentUri tdoc in
+                make_config uri
+              in
+              Mpipeline.make config (Msource.make (Text_document.text tdoc)))
```

Comment:
`Msource.make (Text_document.text tdoc)` is computed twice - lines 148 and 156, is it intentional?

Reply:
It was intentional, but it's actually needlessly complicated. The fact that the earlier `Msource.make` was lazy made it impossible to share across threads. I removed the pointless lazy and now it's possible to share it.